lib/datasource/maven.js

```diff
@@ -0,0 +1,128 @@
+const _ = require('lodash');
+const got = require('got');
+const url = require('url');
+const fs = require('fs-extra');
+const xmlParser = require('fast-xml-parser');
+
+module.exports = {
+  getPkgReleases,
+  resetCache,
+};
+
+let map = new Map();
+
+function resetCache() {
+  map = new Map();
+}
+
+// eslint-disable-next-line no-unused-vars
+async function getPkgReleases(purl, config) {
+  const versions = [];
+  const dependency = getDependencyParts(purl);
+  const repositories = getRepositories(purl);
+  logger.debug(
+    `Found ${repositories.length} repositories for ${dependency.display}`
+  );
+  for (let i = 0; i < repositories.length; i += 1) {
+    const repoUrl = repositories[i];
+    logger.debug(
+      `Looking up ${dependency.display} in repository #${i} - ${repoUrl}`
+    );
+    const mavenMetadata = await downloadMavenMetadata(dependency, repoUrl);
+    if (mavenMetadata) {
+      const newVersions = extractVersions(mavenMetadata).filter(
+        version => !versions.includes(version)
+      );
+      versions.push(...newVersions);
+      logger.debug(`Found ${newVersions.length} new versions for ${dependency.display} in repository ${repoUrl}`); // prettier-ignore
+    }
+  }
+
+  if (versions.length === 0) {
+    logger.warn(`No versions found for ${dependency.display} in ${repositories.length} repositories`); // prettier-ignore
+    return null;
+  }
+  logger.debug(`Found ${versions.length} versions for ${dependency.display}`);
+
+  return {
+    ...dependency,
+    releases: versions.map(v => ({ version: v })),
+  };
+}
+
+function getDependencyParts(purl) {
+  return {
+    display: `${purl.namespace}:${purl.name}`,
+    group: purl.namespace,
+    name: purl.name,
+    version: purl.version,
+    dependencyUrl: purl.namespace.replace(/\./g, '/') + `/${purl.name}`,
+  };
+}
+
+function getRepositories(purl) {
+  return purl.qualifiers.repository_url.split(',').map(url => {
```

Comment:
Add a check for purl.qualifiers && purl.qualifiers.repository_url first (to be safe)